### ISSUE TITLE
Getting pillar error while not using pillar data.

### ISSUE THREAD

**Issue by howardroark** [saltstack/salt#19823]
I am using GitFS in masterless mode with the latest release cut `2014.7.1`.  

Here is the minion file in use:
https://github.com/stackstrap/install/blob/master/files/minion

Here is the gitfs repo:
https://github.com/stackstrap/environment

I get this error when running `salt-call state.highstate`...

```
Data failed to compile:
----------
Pillar failed to render with the following messages:
----------
Rendering Primary Top file failed, render error:
'NoneType' object has no attribute 'endswith'
```

I don't really understand because there is no pillar involved.  I may just be missing something glaring.

Here is the full log: http://pastebin.com/912V11Cw

**Comment by rallytime** [comment]
Interesting. I am not very familiar with using gitfs in a masterless set-up. How did you install salt? Was this working previously and is now now longer working?

**Comment by howardroark** [comment]
So basically it works if the `top.sls` file is part of the `file_roots` on the masterless system.  I can bring in the rest of the states via `gitfs` and they run fine that way.  Only when I move the `top.sls` file to live in one of the `gitfs` repos does the error occur.  It is not a huge deal... I just thought it would be nice to be able to pair different repos in the minion config to get different results.  Kind of a `flow` based configuration style.

Salt is being installed with the `salt-bootstrap` script.

**Comment by rallytime** [comment]
Hey @howardroark - Apologies for not responding here sooner. Have you been able to try this on our latest version of salt (2014.7.5)? Is this still an issue for you? I know a bunch of gitfs fixes have gone in since 2014.7.1.
